clippy_lints/src/matches.rs

```diff
@@ -346,27 +348,28 @@ fn all_ranges(cx: &LateContext, arms: &[Arm]) -> Vec<SpannedRange<ConstVal>> {
     arms.iter()
         .flat_map(|arm| {
             if let Arm { ref pats, guard: None, .. } = *arm {
-                pats.iter()
-            } else {
-                [].iter()
-            }.filter_map(|pat| {
-                if_let_chain! {[
+                    pats.iter()
```

Comment:
This looks weird, can we split it out into `let foo = if let ..` and `foo.filter_map)_` statements?